File: pythonBackend/src/pydantic_models/game.py
from typing import List, Optional
from pydantic import BaseModel, ConfigDict
from datetime import datetime
import uuid

from pydantic_models.Search.SearchDto import BaseSearchDto


class BaseGameDto(BaseModel):
    team1_id: uuid.UUID
    team2_id: uuid.UUID
    sportType: str
    date_played: datetime
    name: str

    model_config = ConfigDict(from_attributes=True)

class CreateGameDto(BaseGameDto):
    ...

class GameDto(BaseGameDto):
    id: uuid.UUID
    team1_name: str
    team2_name: str

class UpdateGameDto(BaseGameDto):
    team1_id: Optional[uuid.UUID]
    team2_id: Optional[uuid.UUID]
    sportType: Optional[str]
    date_played: Optional[datetime]
    name: Optional[str]

class GamesDto(BaseModel):
    games: List[GameDto]

class SearchGameDto(BaseSearchDto):
    filter: str

File: frontend/src/models/game.js
export class Game{
    id;
    team1_id;
    team2_id;
    sport_id;
    date_played;
    constructor(data){
        this.id = data.id;
        this.team1_id = data.team1_id;
        this.team2_id = data.team2_id;
        this.sport_id = data.sport_id;
        this.date_played = data.date_played;
    }
}